### PR TITLE
Allow union types in event properties

### PR DESCRIPTION
**Pull request by badger-adam**
Fixes runtime error when processing events that contain properties with Union Types.

```
Call to undefined method ReflectionUnionType::getName() {"exception":"[object] (Error(code: 0): Call to undefined method ReflectionUnionType::getName() at /home/forge/pay.adammaciuk.com/vendor/
hirethunk/verbs/src/Support/EventStateRegistry.php:132)
```

### CODE REVIEW

**Comment by pitbully** [comment]
Test added for union types. Confirmed test failed prior to the change and succeeded with the changes.

**Comment by badger-adam** [comment]
Test added for union types in events. Confirmed locally that test failed without the changes and succeeded with them.

File: tests/Unit/UseStatesDirectlyInEventsTest.php [hunk 1]
```diff
@@ -101,13 +101,33 @@
     $this->assertEquals($event2->id, $user_request2->last_event_id);
 });
 
+it('supports union typed properties in events', function() {
```
Comment (johncongdon) [inline tests/Unit/UseStatesDirectlyInEventsTest.php:104]: I feel like there be a second test that proves the unhappy path as well.  

Setting the value to a boolean or an array for example.

Comment (badger-adam) [reply tests/Unit/UseStatesDirectlyInEventsTest.php:104]: Isn't that just going to be a PHP TypeError exception?

**Comment by inxilpro** [comment]
Just did review with Daniel. Will merge shortly!

**Pull request merged**